File: Frontend/src/app/relatorios/pex/relatorio-de-solicitacoes/models/modelo.model.ts
export class ModeloModel {
    public id : number = 0;
    public nome: string = "";
    public descricao: string = "";
    public dataIniSolicitacao: string = "";
    public dataFimSolicitacao: string = "";
    public dataIniVencimento: string = "";
    public dataFimVencimento: string = "";
    public colunasRelatorioSelecionadas: number[] = [];
    public escritoriosSelecionados: number[] = [];
    public tiposDeLancamentoSelecionados: number[] = [];
    public ufsSelecionadas: string[] = [];
    public statusSolicitacoesSelecionados: number[] = [];
    public dataDeCadastro: string = "";
    public dataDeAlteracao: string = "";

    public static fromObj(obj: any):ModeloModel {
        var modelo = new ModeloModel();
        modelo.id = obj.id;
        modelo.nome = obj.nome;
        modelo.descricao = obj.descricao;
        modelo.dataIniSolicitacao = obj.dataIniSolicitacao;
        modelo.dataFimSolicitacao = obj.dataFimSolicitacao;
        modelo.dataIniVencimento = obj.dataIniVencimento;
        modelo.dataFimVencimento = obj.dataFimVencimento;
        modelo.colunasRelatorioSelecionadas = obj.colunasRelatorioSelecionadas;
        modelo.escritoriosSelecionados = obj.escritoriosSelecionados;
        modelo.tiposDeLancamentoSelecionados = obj.tiposDeLancamentoSelecionados;
        modelo.ufsSelecionadas = obj.ufsSelecionadas;
        modelo.statusSolicitacoesSelecionados = obj.statusSolicitacoesSelecionados;
        modelo.dataDeAlteracao = obj.dataDeAlteracao;
        modelo.dataDeCadastro = obj.dataDeCadastro;

        return modelo;
    }
}

File: Backend/V2/Oi.Juridico.WebApi.V2/Areas/RelatorioSolicitacaoLancamentoPex/Models/ModeloModel.cs
using Newtonsoft.Json;
using Oi.Juridico.Contextos.V2.RelatorioSolicitacaoLancamentoPexContext.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Oi.Juridico.WebApi.V2.Areas.RelatorioSolicitacaoLancamentoPex.Models
{
    public class ModeloModel
    {
        public decimal Id { get; set; } = 0;
        public string Nome { get; set; } = "";
        public string Descricao { get; set; } = "";
        public string DataIniSolicitacao { get; set; } = "";
        public string DataFimSolicitacao { get; set; } = "";
        public string DataIniVencimento { get; set; } = "";
        public string DataFimVencimento { get; set; } = "";
        public List<int>? ColunasRelatorioSelecionadas { get; set; } = new List<int>();
        public List<int>? EscritoriosSelecionados { get; set; } = new List<int>();
        public List<int>? TiposDeLancamentoSelecionados { get; set; } = new List<int>();
        public List<string>? UfsSelecionadas { get; set; } = new List<string>();
        public List<int>? StatusSolicitacoesSelecionados { get; set; } = new List<int>();
        public string DataDeCadastro { get; set; } = "";
        public string DataDeAlteracao { get; set; } = "";
        public ModeloModel() { }
        public ModeloModel(AgPexRelSolicModelo entity) {
            Id = entity.Id;
            Nome = entity.Nome;
            Descricao = entity.Descricao;
            DataIniSolicitacao = entity.DatIniSolcitacao != null ? entity.DatIniSolcitacao.Value.ToString("dd/MM/yyyy") : "";
            DataFimSolicitacao = entity.DatFimSolcitacao != null ?  entity.DatFimSolcitacao.Value.ToString("dd/MM/yyyy") : "";
            DataIniVencimento = entity.DatIniVencimento != null ? entity.DatIniVencimento.Value.ToString("dd/MM/yyyy") : "";
            DataFimVencimento = entity.DatFimVencimento != null ? entity.DatFimVencimento.Value.ToString("dd/MM/yyyy") : "";
            ColunasRelatorioSelecionadas = String.IsNullOrEmpty(entity.ColunasRelSelecionadas) ? new List<int>() : JsonConvert.DeserializeObject<List<int>>(entity.ColunasRelSelecionadas);
            EscritoriosSelecionados = String.IsNullOrEmpty(entity.EscritoriosRelSelecionados) ? new List<int>() : JsonConvert.DeserializeObject<List<int>>(entity.EscritoriosRelSelecionados);
            TiposDeLancamentoSelecionados = String.IsNullOrEmpty(entity.TiposLancRelSelecionados) ? new List<int>() : JsonConvert.DeserializeObject<List<int>>(entity.TiposLancRelSelecionados);
            StatusSolicitacoesSelecionados = String.IsNullOrEmpty(entity.StatusSolRelSelecionados) ? new List<int>() : JsonConvert.DeserializeObject<List<int>>(entity.StatusSolRelSelecionados);
            UfsSelecionadas = String.IsNullOrEmpty(entity.UfsSelecionadas) ? new List<string>() : JsonConvert.DeserializeObject<List<string>>(entity.UfsSelecionadas);
            DataDeCadastro = entity.DatCriacao != null ? entity.DatCriacao.Value.ToString("dd/MM/yyyy") : "";
            DataDeAlteracao = entity.DatAlteracao != null ? entity.DatAlteracao.Value.ToString("dd/MM/yyyy") : "";
        }
        public AgPexRelSolicModelo retornaEntity()
        {
            if (String.IsNullOrEmpty(Nome))
            {
                throw new Exception("Preencha o nome do modelo");
            }

            return new AgPexRelSolicModelo()
            {
                Nome = Nome,
                Descricao = Descricao,
                DatIniSolcitacao = !String.IsNullOrEmpty(DataIniSolicitacao) ? DateTime.Parse(DataIniSolicitacao) : null,
                DatFimSolcitacao = !String.IsNullOrEmpty(DataFimSolicitacao) ? DateTime.Parse(DataFimSolicitacao) : null,
                DatIniVencimento = !String.IsNullOrEmpty(DataIniVencimento) ? DateTime.Parse(DataIniVencimento) : null,
                DatFimVencimento = !String.IsNullOrEmpty(DataFimVencimento) ? DateTime.Parse(DataFimVencimento) : null,
                ColunasRelSelecionadas = JsonConvert.SerializeObject(ColunasRelatorioSelecionadas),
                EscritoriosRelSelecionados = JsonConvert.SerializeObject(EscritoriosSelecionados),
                TiposLancRelSelecionados = JsonConvert.SerializeObject(TiposDeLancamentoSelecionados),
                UfsSelecionadas = JsonConvert.SerializeObject(UfsSelecionadas),
                UsrCodUsuario = "Perlink",
                StatusSolRelSelecionados = JsonConvert.SerializeObject(StatusSolicitacoesSelecionados),
                DatCriacao = DateTime.Now,
                DatAlteracao = DateTime.Now
            };
        }
    }

} 
